### DISCUSSION TITLE
how to declare k8s services ( kubernetes_sd_configs role service discovery) to probe using blackbox exporter

### DISCUSSION THREAD

**Discussion by sswieton** [prometheus-operator/prometheus-operator#3852] [General]
Hi all,
I'd like to supervise K8s services using blackbox exporter probing, with a dynamically discover of the services to build the targets.

so, I need to be able to use different modules (  to check specific tcp port availibility, HTTP to check URL accessibility ,..). I also need to be able to provision the blackbox exporter targets using a **kubernetes_sd_configs based on role service discovery.**

This is an example of a job I need to create ( this is currently implemented using templates) . ( see below)

I tried these differents objects  provided by the operator but this doesn't match 
        - serviceMonitor implements  kubernetes_sd_configs only for role endpoint and not for role Service
        - Probe implements static probe discovery
        - additional-scrape-configs doesn't provide flexibility

Any  idea ? How could I implement this solution based on the prometheus operator ?
Any advise would be very appreciated.
many thanks for your help
regards

- job_name: 'blackbox'
metrics_path: /probe
      **params:
          module: [tcp_connect]**
        kubernetes_sd_configs:
        - **role: service**
          namespaces:
            names:
               - my-namespace
        relabel_configs:
        - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape]
          action: keep
          regex: true
       ...
        - source_labels: [__meta_kubernetes_service_label_domain]
          action: keep
          regex: myappli
        - source_labels: [__meta_kubernetes_namespace]
          target_label: __address__
          replacement: **blackbox-exporter:9115**  # Blackbox exporter.

**Comment by brancz** [comment]
We only support static targets and discovering via ingress today, but I think using the service role for this makes a lot of sense! I'll mark this as a feature request instead.

Do you want to contribute this?

**Reply by sswieton** [reply]
Thanks a lot for your quick reply and to take into account this feature.
At the moment I do not have the band with to support actively but would be happy to be updated
thanks
Regards

**Reply by frank-berlin** [reply]
+1
